src/provideEndpoints.ts

```diff
@@ -0,0 +1,69 @@
+import { Endpoints } from './models/core/Endpoints';
+import { Environment } from './models/core/Environment';
+import { CloudRegion } from './models/core/CloudRegion';
+import { ServingConfig } from './models/core/SearchConfig';
+
+export const defaultApiVersion = 20220511;
+
+/**
+ * Provides methods for getting various endpoints.
+ *
+ * @internal
+ */
+export class EndpointsAdapter {
```

Comment:
ok I'm sorry about the name again but since we're not sending the endpoints param into this class can we make it an EndpointsFactory instead?

Reply:
Of course. Updated!